src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -964,12 +964,17 @@ private TypeSymbol BindVariableType(CSharpSyntaxNode declarationNode, Diagnostic
                 var constantValueDiagnostics = localSymbol.GetConstantValueDiagnostics(initializerOpt);
                 foreach (var diagnostic in constantValueDiagnostics)
                 {
-                    diagnostics.Add(diagnostic);
-                    hasErrors = true;
+                    localDiagnostics.Add(diagnostic);
+                    if (diagnostic.Severity == DiagnosticSeverity.Error)
+                    {
+                        hasErrors = true;
+                    }
                 }
             }
 
-            diagnostics.AddRangeAndFree(localDiagnostics);
+            diagnostics.AddRange(localDiagnostics.AsEnumerable().Distinct());
```

Comment:
This doesn't seem cheap. Unless it is critical to your PR, you probably shouldn't dedupe like this.

---
In reply to: [126303785](https://github.com/dotnet/roslyn/pull/20729#discussion_r126303785) [](ancestors = 126303785)